pkg/abstractions/secret/http.go

```diff
@@ -0,0 +1,136 @@
+package secret
+
+import (
+	"net/http"
+
+	"github.com/beam-cloud/beta9/pkg/auth"
+	"github.com/beam-cloud/beta9/pkg/types"
+	"github.com/labstack/echo/v4"
+	"github.com/lib/pq"
+)
+
+type secretGroup struct {
+	routeGroup *echo.Group
+	ss         *WorkspaceSecretService
+}
+
+func registerSecretRoutes(g *echo.Group, ss *WorkspaceSecretService) *secretGroup {
+	group := &secretGroup{
+		routeGroup: g,
+		ss:         ss,
+	}
+
+	g.POST("/:workspaceId", auth.WithWorkspaceAuth(group.CreateSecret))
```

Comment:
any reason we're prefixing by workspace ID in the route here? can't we just extract it from the auth token?

Reply:
I'm just thinking we do this so that cluster admins can have access to these endpoints. Maybe it's not a good idea to do it for secrets?

Reply:
Yeah I feel like we should remove the :workspaceId prefix and come up with a different way of "assuming auth" in the cluster admin case

Reply:
would be difficult in future for this particular endpoint. Secret `name` is unique per workspace so workspace would need to be specified in order to access a variable from the cluster admin side. 
A lot of other endpoints use the `/:workspaceId` param